ada_feeding_actionselect/ada_feeding_actionselect/policy_service.py

```diff
@@ -0,0 +1,275 @@
+
+        policy_cls = import_from_string(policy_cls_name)
+        assert issubclass(policy_cls, Policy), f"{policy_cls_name} must subclass Policy"
+
+        kwargs = self.get_kwargs(f"{policy_name}.kws", f"{policy_name}.kwargs")
+
+        # Get the context adapter
+        context_cls_param = self.declare_parameter(
+            "context_class",
+            descriptor=ParameterDescriptor(
+                name="context_class",
+                type=ParameterType.PARAMETER_STRING,
+                description=(
+                    "The class of the context adapter to run, must subclass "
+                    "ContextAdapter. E.g., ada_feeding_actionselect.adapters.NoContext"
+                ),
+                read_only=True,
+            ),
+        )
+
+        context_cls_name = context_cls_param.value
+        if context_cls_name is None:
+            raise ValueError("Context Adapter is required.")
+
+        context_cls = import_from_string(context_cls_name)
+        assert issubclass(
+            context_cls, ContextAdapter
+        ), f"{context_cls_name} must subclass ContextAdapter"
+
+        context_kwargs = self.get_kwargs("context_kws", "context_kwargs")
+        self.context_adapter = context_cls(**context_kwargs)
+
+        # Get the posthoc adapter
+        posthoc_cls_param = self.declare_parameter(
+            "posthoc_class",
+            descriptor=ParameterDescriptor(
+                name="posthoc_class",
+                type=ParameterType.PARAMETER_STRING,
+                description=(
+                    "The class of the posthoc adapter to run, must subclass "
+                    "PosthocAdapter. E.g., ada_feeding_actionselect.adapters.NoContext"
+                ),
+                read_only=True,
+            ),
+        )
+
+        posthoc_cls_name = posthoc_cls_param.value
+        if posthoc_cls_name is None:
+            raise ValueError("Posthoc Adapter is required.")
+
+        posthoc_cls = import_from_string(posthoc_cls_name)
+        assert issubclass(
+            posthoc_cls, PosthocAdapter
+        ), f"{posthoc_cls_name} must subclass PosthocAdapter"
+
+        posthoc_kwargs = self.get_kwargs("posthoc_kws", "posthoc_kwargs")
+        self.posthoc_adapter = context_cls(**posthoc_kwargs)
+
+        # Initialize and validate the policy
+        self.policy = policy_cls(
+            self.context_adapter.dim, self.posthoc_adapter.dim, **kwargs
+        )
+
+        # Start image subscribers
+        self.image = None
+        self.depth = None
+        self.ros_objs = []
+
+        if self.context_adapter.need_rgb:
+            self.ros_objs.append(self.create_subscription(
+                get_img_msg_type("~/image", self), "~/image", self.rgb_callback, 1
+            ))
+        if self.context_adapter.need_depth:
+            self.ros_objs.append(self.create_subscription(
+                get_img_msg_type("~/depth", self), "~/depth", self.depth_callback, 1
```

Comment:
Does this need to be aligned_depth? (i.e., do you envision context adapters making that assumption?) If so, please rename to `~/aligned_depth` to make that clear. (We did this in face and food detection, which needs aligned depth since it assumes the pixels line up.)